src/NaNMath.jl

```diff
@@ -15,4 +15,24 @@ pow(x::Float64, y::Float64) = ccall((:pow,Base.Math.libm),  Float64, (Float64,Fl
 pow(x::Float32, y::Float32) = ccall((:powf,Base.Math.libm), Float32, (Float32,Float32), x, y)
 pow(x,y) = pow(float(x),float(y))
 
+function sum(x::Array{Float64,1})
+    result = NaN
+    for i = x
+        if ~isnan(i)
+            if isnan(result)
+                result = i
+            else
+                result += i
+            end
+        end
+    end
+    if size(x)[1] == 0
+        result = 0.0
+    end
+    if isnan(result)
+        warn("All elements of the array, passed to \"sum\" are NaN!")
```

Comment:
It is not needed, but numpy also prints a warning, and I find it convenient. If something crashes later in the program because of this NaN, then it is nice to see in the console, where it was coming from.